src/components/ComparateurPublicodes/ComparateurPublicodesWidget.tsx

```diff
@@ -0,0 +1,124 @@
+import { fr } from '@codegouvfr/react-dsfr';
+import React from 'react';
+
+import { FormProvider } from '@components/form/publicodes/FormProvider';
+import Heading from '@components/ui/Heading';
+import { Table, type ColumnDef } from '@components/ui/Table';
+import { LocationInfoResponse } from '@pages/api/location-infos';
+import { postFetchJSON } from '@utils/network';
+import { ObjectEntries } from '@utils/typescript';
+
+import { addresseToPublicodesRules, modesDeChauffage } from './mappings';
+import { simulatorTabs } from './Placeholder';
+import useSimulatorEngine from './useSimulatorEngine';
+
+type ComparateurPublicodesWidgetProps = React.HTMLAttributes<HTMLDivElement> & {
+  coords?: [longitude?: number, latitude?: number];
+  city?: string;
+  cityCode?: string;
+  address?: string;
+};
+
+export type TabId = (typeof simulatorTabs)[number]['tabId'];
+const ComparateurPublicodesWidget: React.FC<ComparateurPublicodesWidgetProps> = ({
+  children,
+  coords,
+  city,
+  cityCode,
+  address,
+  ...props
+}) => {
+  const engine = useSimulatorEngine();
+
+  React.useEffect(() => {
+    if (!coords || !city || !cityCode) {
+      return;
+    }
+    const loadInfos = async () => {
+      const infos: LocationInfoResponse = await postFetchJSON('/api/location-infos', {
+        lon: coords[0],
+        lat: coords[1],
+        city,
+        cityCode,
+      });
+
+      engine.setSituation(
+        ObjectEntries(addresseToPublicodesRules).reduce(
+          (acc, [key, infoGetter]) => ({
+            ...acc,
+            [key]: infoGetter(infos) ?? null,
+          }),
+          {}
+        )
+      );
+    };
+    loadInfos();
+  }, [coords, city, cityCode]);
+
+  const modesDeChauffageToDisplay = modesDeChauffage
+    .filter(({ type, label }) => (type as any).includes('collectif') && !label.includes('PAC'))
+    .map((typeInstallation) => ({
+      id: typeInstallation.label,
+      label: typeInstallation.label,
+      bilan: engine.getFieldAsNumber(`Bilan x ${typeInstallation.coutPublicodeKey} . total avec aides`),
+      emissionsCO2: engine.getFieldAsNumber(`env . Installation x ${typeInstallation.emissionsCO2PublicodesKey} . Total`),
+    }))
+    .sort((a, b) => a.bilan - b.bilan);
+
+  const columns: ColumnDef<{ label: string; bilan: number }>[] = [
+    {
+      headerName: 'Mode de chauffage',
+      field: 'label',
+      flex: 3,
+    },
+    {
+      headerName: 'Cout annuel chauffage',
```

Comment:
```suggestion
      headerName: 'Coût annuel chauffage',
```
Pour harmoniser avec le reste